Fix usages of sanitize helper methods for editable content provided by admins

sergei-krylov opened the pull request:
#### :tophat: What? Why?
Fixes usages of sanitize for `description` and `short_description` of `ParticipatoryProcess`, `ParticipatoryProcessGroup` and `ParticipatoryProcessStep`.

#### :pushpin: Related Issues
- Related to #?
- Fixes #9927 

#### Testing
Add video to participatory process `short_description` and `description`, to the participatory group `description`, to the participatory process step `description`.
Then check the corresponding public pages to see that video appears normally.

For participatory process step it is displayed normally before and after this fix, it just switching from `html_safe` to `sanitize` to use same sanitizing way everywhere and be more consistent across views.

ahukkanen commented on decidim-participatory_processes/app/cells/decidim/participatory_processes/content_blocks/highlighted_processes/single_process.erb:
```diff
@@ -14,7 +14,7 @@
                   <h2 class="card__title" id="<%= title_id %>"><%= decidim_html_escape(translated_attribute(process.title)) %></h2>
                 <% end %>
-                <%= decidim_sanitize_editor translated_attribute(process.short_description) %>
+                <%= decidim_sanitize_editor_admin translated_attribute(process.short_description) %>
```
I'm not quite sure if we should allow embeds in the cards. It makes the card really large and not very usable.

Do you have any concrete example where it would be required?

sergei-krylov replied:
I checked it and I agree it almost breaks the layout.
I don't have any requirement to change this. It is reverted in the update.

ahukkanen commented on decidim-participatory_processes/app/views/decidim/participatory_processes/participatory_processes/_promoted_process.html.erb:
```diff
@@ -5,7 +5,7 @@
           <h2 class="card__title"><%= decidim_html_escape(translated_attribute(promoted_process.title)).html_safe %></h2>
         <% end %>
-        <%= decidim_sanitize_editor html_truncate(translated_attribute(promoted_process.short_description), length: 630, separator: "...") %>
+        <%= decidim_sanitize_editor_admin html_truncate(translated_attribute(promoted_process.short_description), length: 630, separator: "...") %>
```
Same thing here. I think showing the embeds in the promoted processes is not necessary.

If you disagree, could you explain a concrete example of this?

sergei-krylov replied:
Same, reverted in update.

ahukkanen commented on decidim-participatory_processes/app/views/decidim/participatory_processes/participatory_processes/_promoted_process_group.html.erb:
```diff
@@ -6,7 +6,7 @@
           <h2 class="card__title"><%= decidim_html_escape(translated_attribute(promoted_process_group.title)).html_safe %></h2>
         <% end %>
-        <%= decidim_sanitize_editor html_truncate(translated_attribute(promoted_process_group.description), length: 630, separator: "...") %>
+        <%= decidim_sanitize_editor_admin html_truncate(translated_attribute(promoted_process_group.description), length: 630, separator: "...") %>
```
Same comment as above.

sergei-krylov replied:
Same, reverted in update.

ahukkanen reviewed (changes requested):
Great fix. I added a couple of questions below, I'm not sure about showing the embeds in the cards.

Also, could you please add few system specs that test the visibility of the embeds on the pages where they were broken? :pray:

sergei-krylov commented:
@ahukkanen, once again - thank you very much for review!
I reverted part of the initial changes, my point was to fix that in description of the participatory process.
I added system tests to cover the changes, as you requested.

sergei-krylov commented:
Just FYI, I checked the failing test locally without introduced changes and it still doesn't work, having the same issue.

ahukkanen reviewed (changes requested):
Great, thanks for the recent changes!

This now looks good regarding the processes but I believe we still have this same bug in other spaces, so I think we should fix those cases too.

I could find the same issue at least at the following locations:

https://github.com/decidim/decidim/blob/2fa06d4bf3c08c1431c3f3a6441ebe7f05fb9ba0/decidim-assemblies/app/views/decidim/assemblies/assemblies/show.html.erb#L42

https://github.com/decidim/decidim/blob/2fa06d4bf3c08c1431c3f3a6441ebe7f05fb9ba0/decidim-conferences/app/views/decidim/conferences/conferences/show.html.erb#L28

https://github.com/decidim/decidim/blob/2fa06d4bf3c08c1431c3f3a6441ebe7f05fb9ba0/decidim-elections/app/cells/decidim/votings/content_blocks/landing_page/description_cell.rb#L19

Could we still fix those and add the same system spec to these as well that you added to processes? :pray: 

---

EDIT: I think the same bug may also exist in consultations but I'm not sure if it should be fixed there because the text is shown inside a `<p class="lead">...</p>` which makes it very large. Also, the consultations module is kind of in a support mode, it will be deprecated at some point anyways.

ahukkanen commented:
> Just FYI, I checked the failing test locally without introduced changes and it still doesn't work, having the same issue.

Don't worry about those broken specs, they should be fixed at #10014. They are broken because of recent update of the accessibility auditing tool and the loose-ended dependency on that.

ahukkanen reviewed (changes requested):
Great!

Tested the following places:
- Process info page
- Process steps page
- Process group page (description)
- Assembly description
- Voting description

One more place where I found we should probably allow the videos:
https://github.com/decidim/decidim/blob/002e1d5114a81dda54be451e3a59a11cfff99ff5/decidim-conferences/app/views/decidim/conferences/conferences/show.html.erb#L23

Because this is shown as normal text:
![Normal text at intro](https://user-images.githubusercontent.com/864340/200655952-9ef524bf-8905-45ba-abe1-3f04b104b15d.png)

ahukkanen reviewed (approved):
Great! :+1: 

If the CI shows green, this is good to be merged.

sergei-krylov commented:
Just changed that for a `short_description` on conference page.

I have some doubts about that change. On that particular page - yes it could fit the video easily as there is a lot of space for short description. From other side, this might became a bit inconsistent at some point, in case we show the video from `short_description` in one place, and [maybe] don't show it in another place (like card, highlighted items, or similar). I mean, as a user, if I put the video into a `short_description` my expectation is that it will show up everywhere, and if somewhere it is not shown then I can consider this is an issue.
I agree it doesn't make sense in cards and highlights as it breaks the layout, maybe it shouldn't display videos from `short_description` anywhere, and in that case, ideally, it shouldn't allow to add a video in editor for short descriptions, but that's another separate thing...

It's just thoughts, we can move forward with the changes if looks good to you. Thanks!

ahukkanen commented:
> I have some doubts about that change. On that particular page - yes it could fit the video easily as there is a lot of space for short description. From other side, this might became a bit inconsistent at some point, in case we show the video from `short_description` in one place, and [maybe] don't show it in another place (like card, highlighted items, or similar). I mean, as a user, if I put the video into a `short_description` my expectation is that it will show up everywhere, and if somewhere it is not shown then I can consider this is an issue.

I don't see this a problem. The card should just show what fits the layout and disable the user to make mistakes editing it (because they will make such mistakes).

> I agree it doesn't make sense in cards and highlights as it breaks the layout, maybe it shouldn't display videos from `short_description` anywhere, and in that case, ideally, it shouldn't allow to add a video in editor for short descriptions, but that's another separate thing...

Exactly, showing these rich elements in the cards does not make much sense.

Here is the only exception because the short description on the page is shown as normal text, so I thought it makes sense to show such rich content there as well.

Hopefully this will be improved during the redesign process. I'll also need to double check this change with #9658 after this one is merged.
